pytorch_lightning/plugins/ddp_sequential_plugin.py

```diff
@@ -0,0 +1,386 @@
+    from fairscale.nn.pipe import rpc as rpc_pipe
+    from fairscale.nn.pipe.pipeline import PipelineStyle
+
+
+class DDPSequentialPlugin(RPCPlugin):
+    def __init__(self,
+                 balance: Optional[List[int]] = None,
+                 microbatches: int = 8,
+                 checkpoint: str = 'except_last',
+                 balance_mode: str = "balance_by_size",
+                 pipelined_backward: Optional[bool] = True,
+                 **kwargs):
+        """
+        Provides sequential model parallelism for :class:`nn.Sequential <torch.nn.Sequential>` module.
+        If the module requires lots of memory, Pipe can be used to reduce this by leveraging multiple GPUs.
+        ::
+            Example::
+                class MyLightningModule:
+                    def __init__(self):
+                        ...
+                        model.sequential_module = torch.nn.Sequential(my_layers)
+
+                # Split my module across 4 gpus, one layer each
+                model = MyLightningModule()
+                plugin = DDPSequentialPlugin(balance=[1, 1, 1, 1])
+                trainer = Trainer(accelerator='ddp', gpus=4, plugins=[plugin])
+                trainer.fit(model)
+
+        .. _DDPSequentialPlugin: https://arxiv.org/abs/1811.06965
+
+        Pipeline parallelism comes with with checkpointing to reduce peak
+        memory required to train while minimizing device under-utilization.
+        This is turned on by default and can be turned off via the checkpoint argument.
+
+        You should determine the balance when defining the plugin,
+        or you can pass an example input array via the LightningModule to infer a balance.
+        The module will be partitioned into multiple devices according to the given balance. You may also rely on
+        your own heuristics to find your own optimal configuration.
+        Args:
+            balance: The balance of the model, i.e [2, 2] (two layers on each GPU).
+            If not provided assumes user provides an input example array to find a balance on all GPUs.
+            microbatches: Allows for parallelization to reduce device utilization
+            by splitting the batch into further smaller batches.
+            checkpoint: Enables gradient checkpointing. ['always', 'except_last', 'never']
+            balance_mode: Type of balance heuristic to use if balance to be inferred.
+            'balance_by_size': checks memory usage of each layer and determines balance,
+            'balance_by_time': checks time of each layer and determines balance
+            pipelined_backward: if True, call torch.autograd.backward once per microbatch on the
+            backward pass (instead of once for the whole batch). This works
+            around a potential deadlock in pytorch when using tensor parallelism
+            at the same time. Defaults to `True` if
+            `get_model_parallel_world_size() > 1`
+            (default: `None`)
```

Comment:
```suggestion

        Args:
            balance: The balance of the model, i.e `[2, 2]` (two layers on each GPU).
                If not provided assumes user provides an input example array to find a balance on all GPUs.
            microbatches: Allows for parallelization to reduce device utilization
                by splitting the batch into further smaller batches.
            checkpoint: Enables gradient checkpointing. Options: `['always', 'except_last', 'never']`
            balance_mode: Type of balance heuristic to use if balance to be inferred.
            'balance_by_size': checks memory usage of each layer and determines balance,
            'balance_by_time': checks time of each layer and determines balance
            pipelined_backward: if True, call ``torch.autograd.backward`` once per microbatch on the
                backward pass (instead of once for the whole batch). This works
                around a potential deadlock in pytorch when using tensor parallelism
                at the same time. Defaults to `True` if `get_model_parallel_world_size() > 1`
                (default: `None`)
```